api/response/schema.py

```diff
@@ -16,26 +17,37 @@ class Meta:
 
 class CreateResponse(graphene.Mutation):
     class Arguments:
-        question_id = graphene.Int(required=True)
+        question_id = graphene.List(graphene.Int, required=True)
         room_id = graphene.Int(required=True)
         rate = graphene.Int()
         text_area = graphene.String()
         missing_items = graphene.List(graphene.Int)
 
-    response = graphene.Field(Response)
+    response = graphene.List(Response)
 
     def mutate(self, info, **kwargs):
         validate_empty_fields(**kwargs)
         query = Room.get_query(info)
         room = query.filter_by(id=kwargs['room_id']).first()
         if not room:
             raise GraphQLError("Non-existent room id")
-        get_question = Question.query.filter_by(id=kwargs['question_id']).first() # noqa
-        if not get_question:
-            raise GraphQLError("Question does not exist")
-        question_type = get_question.question_type
-        resource = create_response(question_type, **kwargs)
-        return CreateResponse(response=resource)
+        question_ids = kwargs.pop('question_id')
+        resources = []
+        questions = Question.get_query(info).filter(
+            QuestionModel.id.in_(question_ids)).all()
+        # checks if the number of questions found equals
+        # the number of question ids sent in the mutation
+        if len(questions) == len(question_ids):
+            for question in questions:
+                question_type = question.question_type
+                resource = create_response(
+                    question_type, question_id=question.id, **kwargs)
+                resources.append(resource)
+        else:
+            raise GraphQLError(
+                "One or more of the provided questions does not exist"
```

Comment:
@kampkelly, slight concern, if you are checking if the questions that the user has input exist, is checking that using the length a good idea? The lengths could be similar but the question ids also inexistent. For example existing questions are [1,2,3] and I input [3,4,5], length is still the same but they don't exist. From what I'm seeing this scenario doesn't raise this error. Kindly correct me if I'm wrong.

Reply:
looking at @bevkololi 's concern I think the use of sets could make it better. i.e `question_ids=[1,2,3,4,5,6,7,8,9,0]` and `question_id =[4,5,6]`
then:
`set(question_ids)&set(question_id)` returns `[4,5,6]`

Reply:
I agree with @dnuwa on this.

Reply:
True @dnuwa. Or simply looping through the argument and checking if it exists in the question_id list.

Reply:
I also have another concern. When validating the data passed in, is there a way to validate that for each question passed,  the response is of the type the question expects. For instance if I pass two questions that expect a `rate` type and pass only one rate type it will give both questions that same rating.

Reply:
@caveinn Thanks for the feedback. I see your concern but that may have to be handled in another story. My task is to allow saving multiple questions at a time and not to validate the question_type.

Reply:
These validations already existed @kampkelly, so it'll just be finding a way to incorporate them to this new structure of response mutations.

Reply:
@bevkololi  and @dnuwa thanks for your feedback. But I belive checking with the length will work as expected. This is because the code on line 36 in api/response/schema.py checks if there are questions with the id's sent in the mutation and returns a list of this questions if found.
So if you pass id's [3,4,5] of which two id's do not exist. It returns a list of count 1.
So when I check if the length is the same with the input sent (count 3). This will not be equal and so the error will be raised. 
I can explain this to everyone during the backend sync today.

Reply:
I would suggest  that you be more specific with your error messages.

Reply:
Thanks @Bonifase I will work on that.

Reply:
This has been fixed now.